### ISSUE TITLE
Proposal about remove script blocking for better performance

### ISSUE THREAD

**Issue by kuitos** [marko-js/writable-dom#6]
### Description
Hi there👋
Thanks for your great works about such an elegant library, I have greatly benefited from it.

Recently I discovered a small issue while reading the source code, which is that script insertion is also blocking. https://github.com/marko-js/writable-dom/blob/020d50b97d197334c8e6609eb949a0e6a361706e/src/index.ts#L156-L162

This behavior is somewhat inconsistent with the browser's behavior. When parsing a series of non-async scripts, the browser could automatically ensure that their execution follows the order of the DOM, without requiring us to manually control it through blocking. Manual control should only be necessary for CSS loading.

### Why
Theoretically, this fix would be a certain improvement in performance.

### Possible Implementation & Open Questions
Just removing the blocking check for scripts should be enough, unless there are some special scenarios that I am not aware of.

### Is this something you're interested in working on?

Yes

**Comment by DylanPiercey** [comment]
Could you explain how you expect the execution to be? Blocking scripts need to be in executed In order after any previously rendered style sheets and scripts, which is what this implementation does. To my knowledge this works the same as browsers.

If you want scripts to execute eagerly you can use async, or module scripts.

**Comment by kuitos** [comment]
> Blocking scripts need to be in executed In order after any previously rendered style sheets and scripts

Yes it is, but for blocking scripts browsers could ensure the executing order without any manual controlling.

Here are the examples that show the differences between them:

#### dynamic scripts load in browsers
1. we have two external script where `a.js` downloads with 1s delay but b.js without ayn delay
```js
// a.js with 1s delay while downloading
console.log('a');

// b.js
console.log('b');
```
2. we simulate the behavior of the browser with parallel inserts
```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Title</title>
</head>
<body>

<script>
  const scriptA = document.createElement('script')
  scriptA.src = '/a.js';
  scriptA.async = false;

  const scriptB = document.createElement('script')
  scriptB.src = '/b.js';
  scriptB.async = false;

  // parallel insert script to document
  document.body.appendChild(scriptA);
  document.body.appendChild(scriptB);

</script>
</body>
</html>
```
3. we can find that the console logs `a` then `b` in orders although b.js downloaded first
![image](https://github.com/marko-js/writable-dom/assets/5206843/9d496ee8-ffff-4904-9918-f1b58526c780)
![image](https://github.com/marko-js/writable-dom/assets/5206843/2cd77044-3e52-44b8-80b6-134945b0a89f)

We don't need to manual control the executing order between a.js and b.js, browsers could automatically guarantee their execution follows the order of DOM inserting, when the script element async attribute set as false.

#### In writable-dom implementation, these scripts work like that:
```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Title</title>
</head>
<body>

<script>
  const scriptA = document.createElement('script')
  scriptA.src = '/a.js';
  scriptA.async = false;
  
  // after a.js downloaded and executed then we insert script b.js to document
  scriptA.onload = () => {
      const scriptB = document.createElement('script')
      scriptB.src = '/b.js';
      scriptB.async = false;
      document.body.appendChild(scriptB);
  }

  document.body.appendChild(scriptA);
</script>
</body>
</html>
```
scriptB must wait until scriptA downloaded and executed then it could be inserted into document.


Here I made [a codesandbox example](https://codesandbox.io/p/sandbox/dynamic-script-5t2vfs?file=/src/index.html:1,1) to explain what I mean, pls take a look😃

**Comment by DylanPiercey** [comment]
Ah I think you're missing that writable dom is two phase. It inserts preloads while blocked for any down the line assets. So in your scenario, even when `b` is blocked by `a`, `b` is still downloaded as soon as possible (just not executed).

**Comment by kuitos** [comment]
I actually do know about the preload mechanism😄.

But just now, I realized I made a mistake. I confused dynamic script loading with the normal HTML parsing script flow. In dynamic script loading scenarios, async false can ensure both execution order and earliest possible insertion. But in the standard HTML document rendering flow, synchronous scripts will block further document parsing, which is consistent with the implementation of writable-dom.

Thank you for your patient explanation, I am closing this issue~
